src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -408,9 +408,13 @@ protected virtual async Task ExecuteNodeAsync(ExecutionContext context, Executio
                 if (result is not IDataLoaderResult)
                 {
                     CompleteNode(context, node);
-                    // for non-dataloader nodes that completed without throwing an error, we can re-use the context
-                    resolveContext.Reset(null, null);
-                    System.Threading.Interlocked.CompareExchange(ref context.ReusableReadonlyResolveFieldContext, resolveContext, null);
+                    // for non-dataloader nodes that completed without throwing an error, we can re-use the context,
+                    // except for enumerable lists - in case the user returned a value with LINQ that is based on the context source
+                    if (result is not IEnumerable)
```

Comment:
One may have such resolver:
```
context => new MyObject { Values = context.Source.Where(x => x.Age > 42); }
```

Reply:
The question of the context usage is much wider than just checking the return type.

Reply:
Or `context => new MyObject(context.Source.Where(x => x.Age > 42));`

Reply:
I am not sure that it makes sense to consider a separate case and adjust the code under it, especially since not 100% of cases of change will be useful. I can return `IEnumerable` without the use of context:
`context => Enumerable.Range(5);`

Reply:
We have `Copy` method with clear semantic. It will work for any situation that I demonstated above.

Reply:
I doubt it makes sense to take these changes. They are not full. The problem is not solved, questions remain.

Reply:
The only way to 'fully' fix context re-use is to not re-use the context.

This change covers an EXTREMELY COMMON use case: returning a LINQ IEnumerable from a field resolver for a ListGraphType.  It is also an explicit change, specifically designed to support list graph types.  And, it does fix ALL list graph types (since they are ALL required to return an IEnumerable).  It is not intended to fix issues with object graph types, nor does it do so.

We should not have code that breaks for normal use cases, even if such behavior is documented.

Reply:
I understand. Well, OK, let this common case be considered explicitly.